csrc/multidevice/lower_communication.cpp

```diff
@@ -538,18 +539,28 @@ std::vector<std::shared_ptr<Communication>> lowerCommunication(
 }
 
 bool isLowerableToCommunication(Expr* expr) {
+  NVF_ERROR(
+      ir_utils::isTvOp(expr),
+      "Non-tv op is not supported yet: ",
+      expr->toString());
   if (expr->isA<ReductionOp>()) {
-    auto out = expr->as<ReductionOp>()->out();
-    NVF_ERROR(out->isA<TensorView>(), "output is not a TensorView");
-    auto out_tv = out->as<TensorView>();
+    auto in = expr->as<ReductionOp>()->in()->as<TensorView>();
+    auto out = expr->as<ReductionOp>()->out()->as<TensorView>();
+    // get the reduced axis
     std::vector<IterDomain*> reduction_axis;
     std::copy_if(
-        out_tv->getMaybeRFactorDomain().begin(),
-        out_tv->getMaybeRFactorDomain().end(),
+        out->getMaybeRFactorDomain().begin(),
```

Comment:
```suggestion
        out->getRootDomain().begin(),
```

It doesn't actually matter as there should be no rfactor here, but semantically when a consumer tensor is matched up with a producer tensor, the root domain of the consumer should be used.

Reply:
ok thanks that's helpful for my understanding